public/app/features/dashboard-scene/panel-edit/PanelVizTypePicker.tsx

```diff
@@ -21,25 +27,27 @@ export interface Props {
 }
 
 export function PanelVizTypePicker({ vizManager, data, onChange }: Props) {
-  const { panel } = vizManager.useState();
+  const { panel, sourcePanel } = vizManager.useState();
   const styles = useStyles2(getStyles);
   const [searchQuery, setSearchQuery] = useState('');
 
   const isWidgetEnabled = false;
   const tabKey = isWidgetEnabled ? LS_WIDGET_SELECT_TAB_KEY : LS_VISUALIZATION_SELECT_TAB_KEY;
   const defaultTab = isWidgetEnabled ? VisualizationSelectPaneTab.Widgets : VisualizationSelectPaneTab.Visualizations;
   const panelModel = useMemo(() => new PanelModelCompatibilityWrapper(panel), [panel]);
+  const vizPanel = sourcePanel.resolve();
+  const oldPanelModel = getOldPanelModel(vizPanel);
```

Comment:
@mdvictor - we should not use the legacy model if possible. I suggest using `PanelModelCompatibilityWrapper` which is a PanelModel API around a `VizPanel`. Here's an example how to use it: https://github.com/grafana/grafana/blob/main/public/app/features/dashboard-scene/panel-edit/PanelVizTypePicker.tsx#L31

Reply:
I'll look more into this. Think we need to extend the panelModel to support the library panel prop `libraryPanel?: LibraryPanelRef | LibraryPanel;`. The `PanelLibraryOptionsGroup` component uses the PanelModel from `dashboard/state` and not the one from `grafana/data` which has these library details.